staging/src/k8s.io/kubectl/pkg/kuberc/kuberc.go

```diff
@@ -111,9 +113,64 @@ func (p *Preferences) Apply(rootCmd *cobra.Command, args []string, errOut io.Wri
 	if err != nil {
 		return args, err
 	}
+	err = p.applyAllowlist(cfg, kuberc)
+	if err != nil {
+		return args, err
+	}
+
 	return args, nil
 }
 
+// applyAllowlist compares the configured credential plugin against all
+// criteria for each entry in the allowlist
+func (p *Preferences) applyAllowlist(cfg clientcmd.ClientConfig, kuberc *config.Preference) error {
+	if kuberc.CredPluginAllowlist == nil {
+		return nil
+	}
+
+	allowlist := *kuberc.CredPluginAllowlist
+
+	rcfg, err := cfg.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	if rcfg.ExecProvider == nil || rcfg.ExecProvider.Command == "" {
```

Comment:
```suggestion
	if rcfg.ExecProvider == nil || len(rcfg.ExecProvider.Command) == 0 {
```

Applies to other places as well -> kube convention is to compare length.

Reply:
Applies to other places? I only see the one place

Reply:
Nevermind, I see other places

Reply:
done